include/fi_util.h

```diff
@@ -295,6 +295,28 @@ struct util_event {
 int fi_eq_create(struct fid_fabric *fabric, struct fi_eq_attr *attr,
 		 struct fid_eq **eq_fid, void *context);
 
+/*
+ * Buffer Pool
+ */
+
+struct util_buf_pool {
+	size_t size;
+	size_t num;
+	void *memory_region;
+	struct slist list;
+};
+
+struct util_buf {
+	struct slist_entry entry;
+	struct util_buf_pool *pool;
+	uint8_t data[0];
```

Comment:
@bturrubiates - Switched to data[] here. But looks like we have a similar instance in the same file. Will get it cleaned too.